blocks/lists.js

```diff
@@ -264,20 +272,72 @@ Blockly.Blocks['lists_create_with'] = {
         // but with an empty default value
         var newInput = this.getInput('ADD' + (this.itemCount_ - 1));
         var shadowInputDom = firstInput.connection.getShadowDom();
-        if (shadowInputDom) {
-          var shadowDom = Blockly.utils.xml.createElement('shadow');
-          var shadowInputType = shadowInputDom.getAttribute('type');
-          shadowDom.setAttribute('type', shadowInputType);
-          var shadowDomField = Blockly.utils.xml.createElement('field');
-          shadowDomField.setAttribute('name', 'NUM');
-          shadowDom.appendChild(shadowDomField);
-          if (shadowDom) {
-            shadowDom.setAttribute('id', Blockly.utils.genUid());
-            newInput.connection.setShadowDom(shadowDom);
-            newInput.connection.respawnShadow_();
+        var newShadowType = shadowInputDom && shadowInputDom.getAttribute('type');
+        if (newShadowType) {
+          var shadowDom = createBlockDom('shadow', newShadowType);
+          var childValues = getChildrenByTag(shadowInputDom, 'value');
+
+          for (var i = 0; i < childValues.length; i++) {
+            var input = childValues[i];
+            var inputName = input.getAttribute('name');
+            var fieldShadow = getChildrenByTag(input, 'shadow')[0];
```

Comment:
should probably call this "valueShadow" (same below) to match the blockly terminology